components/style/properties/longhand/font.mako.rs

```diff
@@ -474,7 +477,7 @@ ${helpers.single_keyword_system("font-variant-caps",
 
             /// Convert from an integer to Weight
             pub fn from_int(n: i32) -> Result<Self, ()> {
-                if n >= 100 && n <= 900 && n % 100 == 0 {
+                if n >= 0 && n <= u16::MAX {
```

Comment:
Parsing also uses this function, so this change would make it accept values it is not currently supposed to accept.

I guess it is probably better to just leave this function as is, and change `from_gecko_weight` to accept a wider range.